pandas/io/parsers/python_parser.py

```diff
@@ -560,12 +556,26 @@ def _infer_columns(
                 )
             else:
                 # Ignore output but set used columns.
-                self._handle_usecols([names], names, ncols)
                 columns = [names]
-                num_original_columns = ncols
+                self._handle_usecols(columns, columns[0], ncols)
 
         return columns, num_original_columns, unnamed_cols
 
+    @cache_readonly
+    def _header_line(self):
+        # Store line for reuse in _get_index_name
+        if self.header is not None:
+            return None
```

Comment:
Won't `ncols = len(self._header_line)` fail if this is hit?

Reply:
assuming you're referring to L543, it would, but the "else" on L542 ensures that we dont get there with self.header not-None.